### PR TITLE
feat(swarm): log local peer id on swarm creation

### LINKED ISSUE #4394: feat(swarm): print local `PeerId` on `log::info!` level
We currently repeat the line below across all our examples and binaries (e.g. `libp2p-server`):

https://github.com/libp2p/rust-libp2p/blob/c6984dd51db6be8e260be1494f2bbc9052a42206/examples/chat-example/src/main.rs#L49

Instead, let's print it on creation or first `poll` of `Swarm`.

Suggested by @thomaseizinger in https://github.com/libp2p/rust-libp2p/pull/4120#discussion_r1292733378.

### PR DESCRIPTION
**Pull request by sufferpurrityxd**
## Description
Closes https://github.com/libp2p/rust-libp2p/issues/4394.

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: swarm/src/lib.rs [hunk 1]
```diff
@@ -1520,6 +1520,7 @@ where
 
     /// Builds a `Swarm` with the current configuration.
     pub fn build(self) -> Swarm<TBehaviour> {
+        log::info!("Local peer id: {}", self.local_peer_id);
```
Comment (thomaseizinger) [inline swarm/src/lib.rs:1523]: Do all paths go through this `build` method or is it possible to construct a `Swarm` without calling this?

**Review by thomaseizinger — CHANGES_REQUESTED** [review]
Thank you! :)

Looks like you've missed a few print statements:

```
❯ rg '!\("Local peer id'
misc/server/src/main.rs
82:    info!("Local peer id: {local_peer_id}");

libp2p/src/tutorials/ping.rs
80://!     println!("Local peer id: {local_peer_id:?}");
114://!     println!("Local peer id: {local_peer_id:?}");
153://!     println!("Local peer id: {local_peer_id:?}");
191://!     println!("Local peer id: {local_peer_id:?}");
248://!     println!("Local peer id: {local_peer_id:?}");
298://!     println!("Local peer id: {local_peer_id:?}");

swarm/src/lib.rs
1523:        log::info!("Local peer id: {}", self.local_peer_id);

examples/rendezvous/src/main.rs
57:    log::info!("Local peer id: {}", swarm.local_peer_id());

examples/rendezvous/src/bin/rzv-identify.rs
59:    log::info!("Local peer id: {}", swarm.local_peer_id());

examples/rendezvous/src/bin/rzv-discover.rs
59:    log::info!("Local peer id: {}", swarm.local_peer_id());

examples/rendezvous/src/bin/rzv-register.rs
60:    log::info!("Local peer id: {}", swarm.local_peer_id());

examples/metrics/src/main.rs
44:    info!("Local peer id: {local_peer_id:?}");
```

File: swarm/src/lib.rs (hunk 1 above)
Comment (sufferpurrityxd) [reply swarm/src/lib.rs:1523]: No, `Swarm` is only created through `SwarmBuilder` and only after calling the `build` function

**Comment by sufferpurrityxd** [comment]
@thomaseizinger I forgot that there is also `log::info!("Local peer id....")`, removed in all places except for the `tutorial`, it seems to me that it does not need to be removed there, if this still needs to be done, then I will make changes.

**Comment by thomaseizinger** [comment]
> @thomaseizinger I forgot that there is also `log::info!("Local peer id....")`, removed in all places except for the `tutorial`, it seems to me that it does not need to be removed there, if this still needs to be done, then I will make changes.

Please remove them there as well but make sure that we initialize a logger too so people will still see the message :)

File: swarm/src/lib.rs (hunk 1 above)
Comment (thomaseizinger) [reply swarm/src/lib.rs:1523]: @eserilev: This PR is likely going to be merged before your one (https://github.com/libp2p/rust-libp2p/pull/4282). I think we may want to add a "debug" span to `Swarm::poll` too that embeds the local peer ID.

**Comment by sufferpurrityxd** [comment]
Removed all local peer_id prints from tutorial,
corrected some documentation. Maybe it's worth adding recommendations on setting RUST_LOG=info before starting the tutorial?

File: libp2p/src/tutorials/ping.rs [hunk 2]
```diff
@@ -75,17 +76,14 @@
...
 //!     let local_key = identity::Keypair::generate_ed25519();
 //!     let local_peer_id = PeerId::from(local_key.public());
-//!     println!("Local peer id: {local_peer_id:?}");
 //!
 //!     Ok(())
 //! }
 //! ```
 //!
-//! Go ahead and build and run the above code with: `cargo run`. A unique
-//! [`PeerId`](crate::PeerId) should be displayed.
```
Comment (thomaseizinger) [inline libp2p/src/tutorials/ping.rs]: This is a bit odd now, isn't it? Why don't we retain the `println` here until we initalize the `Swarm` and explain in the tutorial that this is being printed implicitly? (And also only then add the logger?). I am open to other suggestions but we should be consistent / make sense here :)

Comment (sufferpurrityxd) [reply libp2p/src/tutorials/ping.rs]: Yes, I agree it looks weird. I decided to return all `println` before the creation of `Swarm,` also added documentation and explained why we remove `println`

Comment (thomaseizinger) [reply libp2p/src/tutorials/ping.rs]: Awesome, thanks!

**Review by thomaseizinger — DISMISSED** [review]
Thank you!

**Comment by drHuangMHT** [comment]
This PR seems to be stalled @thomaseizinger

**Comment by thomaseizinger** [comment]
@sufferpurrityxd There seems to be an error for a broken intra-doc link, see https://github.com/libp2p/rust-libp2p/actions/runs/6099457681/job/16567183946?pr=4439. Mind taking a look at that?

**Comment by sufferpurrityxd** [comment]
I didn't notice this error, fixed it

**Review by thomaseizinger — DISMISSED** [review]

**Review by thomaseizinger — DISMISSED** [review]

**Review by thomaseizinger — APPROVED** [review]

**Pull request merged**